src/app.md

```diff
@@ -234,6 +235,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-file-sizes]: #ex-file-sizes
+<a name="ex-file-sizes"></a>
+## Recursively calculate file sizes at given depth
+
+Combine functional thinking and metadata to calculate total file size at given depth and path (optional).
+
+[![walkdir-badge]][walkdir] [![cat-filesystem-badge]][cat-filesystem]
+
+```
+extern crate walkdir;
+
+use std::env;
+use walkdir::WalkDir;
+
+fn main() {
+    // Get command line arguments
+    let mut env_vars = env::args();
+
+    // Skip first one
+    env_vars.next();
+
+    // Get depth and path or use default values
+    let depth = env_vars.next().unwrap_or("3".to_string())
+                .parse::<usize>().unwrap_or(3);
+    let path = env_vars.next().unwrap_or(".".to_string());
+
+    println!("{}, {}", depth, path);
+
+    let total_size: u64 = WalkDir::new(path)
```

Comment:
Yeah, I know. I'm used to write data types explicitly, because then IDE gives better suggestions.